poo/abstraction/en/Abstraction.md

```diff
@@ -0,0 +1,117 @@
+        System.out.println("Un humain souhaite se déplacer !");
+    }
+
+    void consumeEnergy(float value) {
+        this.energie -= value;
+    }
+}
+```
+
+Now, imagine that we have a class Hiking which allows organise a walk enter Humans. A thing of style:
+
+```java
+class Hiking {
+
+    List<Human> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Human participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Human human : participants) {
+            human.move();
+            human.consumeEnergy(3);
+        }
+    }
+}
+```
+
+So far so good, we have that we want. This code works perfectly, except, in reality, he contain two major problems.
+
+1. **We restraint the hiking of the objects of type Human**, whereas anything can walk and exhausted could take a hiking. It’s not really the kind to forbid to elephants and to tractors to participate :(
+2. We ask a human in full but us all we need, this is his legs (hum). This class hiking, however currently access to the human respiratory system (via `breathe()` ) and to this stomach (via `eat()` ) ! **So, we give too much power to this class Hiking, since she is able to do things who shouldn’t be her accessible.**
+
+So, how to contour this two problems? 
+
+So, now, our class Human will be able to implement this super interface, and in reality we add `implements Walking` and put a little `@Override` on their methods `move` and `consumeEnergy` and the turn is played, since this is a behaviour that we had already implemented beforehand, just without the interface.
+
+```java
+class Human implements Walking {
+
+    int air;
+    int satiety;
+    float energy;
+
+
+    void breathe() {
+        air += 10;
+    }
+
+    void eat() {
+        satiety += 5;
+    }
+
+    @Override
+    void move() {
+        System.out.println("Un humain souhaite se déplacer !");
+    }
+
+    @Override
+    void consumeEnergy(float value) {
+        this.energy -= value;
+    }
+}
+```
+But, what to change? Now, instead of making a human list in our Hiking, we will do a *Walking list* (it’s more politically correct). And, there is almost nothing to change, it’s a good sign:
+
+```java
+class Hiking {
+
+    List<Walking> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Walking participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Walking walking: participants) {
+            walking.move();
+            walking.consumeEnergy(3);
+        }
+    }
+}
+```
+
+And, henceforth, not only our Hiking class hasn’t access to the Human methods `breathe` and `eat`, since, nowhere, we said to our work with humans (no, but think, if a hiking can make you swallow peas, it would be creepy, still), but more, she accepts the elephants and the tractors! Provided that implements `Walking` them too… and, who know how to out of print a tractor.

```

Comment:
Bad pronoun for the class (`she` -> `it)
Fix `that` -> `they`
`
```suggestion
And, henceforth, not only our Hiking class hasn’t access to the Human methods `breathe` and `eat`, since, nowhere, we said to our work with humans (no, but think, if a hiking can make you swallow peas, it would be creepy, still), but more, it accepts the elephants and the tractors! Provided they implement `Walking` them too… and, who know how to out of print a tractor.

```

Reply:
Corrected!

Thank you for your time and correction!
How was my translation? Very bad or can it pass?
Do you think that if I translate another document it will be beneficial or a waste of time, especially for you? :)

Reply:
It's a good translation overall. Even in French documents, we have several comments to make. 
Your translation is really correct. And even with the typo, it will always be beneficial.

Reply:
Ok! Very thank, have a good evening!